internal/generators/scan/cluster/cluster_scanner.go

```diff
@@ -322,7 +323,15 @@ func (cs *ClusterScanner) scanKafkaResources(clusterInfo *types.ClusterInformati
 
 	clientBrokerEncryptionInTransit := types.GetClientBrokerEncryptionInTransit(clusterInfo.Cluster)
 
-	admin, err := cs.kafkaAdminFactory(brokerAddresses, clientBrokerEncryptionInTransit)
+	var kafkaVersion string
+	if clusterInfo.Cluster.Provisioned != nil {
+		kafkaVersion = utils.ConvertKafkaVersion(clusterInfo.Cluster.Provisioned.CurrentBrokerSoftwareInfo.KafkaVersion)
+	} else {
+		slog.Warn("⚠️ Serverless clusters return nil for Kafka version, defaulting to 4.0.0")
+		kafkaVersion = "4.0.0"
+	}
```

Comment:
```suggestion

func (cs *ClusterScanner) getKafkaVersion(clusterInfo *types.ClusterInformation) string {
	if clusterInfo.Cluster.ClusterType == kafkatypes.ClusterTypeServerless {
		return "4.0.0"
	}

	if clusterInfo.Cluster.ClusterType == kafkatypes.ClusterTypeProvisioned {
		return utils.ConvertKafkaVersion(clusterInfo.Cluster.Provisioned.CurrentBrokerSoftwareInfo.KafkaVersion)
	}

	return "4.0.0"
}
```

would something like this be more explicit about whats going on here?

Reply:
or even, note I haven't tested either :P

```go
func (cs *ClusterScanner) getKafkaVersion(clusterInfo *types.ClusterInformation) string {
	switch clusterInfo.Cluster.ClusterType {
	case kafkatypes.ClusterTypeServerless:
		return "4.0.0"
	case kafkatypes.ClusterTypeProvisioned:
		return utils.ConvertKafkaVersion(clusterInfo.Cluster.Provisioned.CurrentBrokerSoftwareInfo.KafkaVersion)
	default:
		slog.Warn("⚠️ Unknown cluster type, defaulting to 4.0.0")
		return "4.0.0"
	}
}
```

Reply:
Yea that's smarter 🤓